### ISSUE TITLE
Alias output redirect issue - I/O operation on closed file

### ISSUE THREAD

**Issue by mitnk** [xonsh/xonsh#2222]
This is a test case in tests/test_integrations.py.
```
def _f(args, stdout):
    print('Wow Mom!', file=stdout)
aliases['f'] = _f
```
With this, run `f o>e` several times would reproduce the following error:
```
$ f o>e
$ f o>e
$ f o>e
Traceback (most recent call last):
  File "/usr/local/lib/python3.5/site-packages/xonsh/base_shell.py", line 322, in default
    run_compiled_code(code, self.ctx, None, 'single')
  File "/usr/local/lib/python3.5/site-packages/xonsh/codecache.py", line 65, in run_compiled_code
    func(code, glb, loc)
  File "<xonsh-code>", line 1, in <module>
  File "/usr/local/lib/python3.5/site-packages/xonsh/built_ins.py", line 858, in subproc_captured_hiddenobject
    return run_subproc(cmds, captured='hiddenobject')
  File "/usr/local/lib/python3.5/site-packages/xonsh/built_ins.py", line 826, in run_subproc
    command.end()
  File "/usr/local/lib/python3.5/site-packages/xonsh/proc.py", line 1991, in end
    self._end(tee_output=tee_output)
  File "/usr/local/lib/python3.5/site-packages/xonsh/proc.py", line 1999, in _end
    for _ in self.tee_stdout():
  File "/usr/local/lib/python3.5/site-packages/xonsh/proc.py", line 1915, in tee_stdout
    for line in self.iterraw():
  File "/usr/local/lib/python3.5/site-packages/xonsh/proc.py", line 1797, in iterraw
    stdout = NonBlockingFDReader(stdout.fileno(), timeout=timeout)
ValueError: I/O operation on closed file
```
Both Linux & Mac have this issue.

**Comment by scopatz** [comment]
Thanks for reporting @mitnk. I thought I had fixed that... oh well.

**Comment by scopatz** [comment]
Potential fix in #2559

**Comment by gforsyth** [comment]
Closed from #2559
